tablecodec/tablecodec.go

```diff
@@ -227,6 +227,60 @@ func DecodeColumnValue(data []byte, ft *types.FieldType, loc *time.Location) (ty
 	return colDatum, nil
 }
 
+// DecodeRowWithMap decodes a byte slice into datums with a existing row map.
+// Row layout: colID1, value1, colID2, value2, .....
+func DecodeRowWithMap(b []byte, cols map[int64]*types.FieldType, loc *time.Location, row map[int64]types.Datum) error {
```

Comment:
There are duplicated code in `DecodeRow`, we can make `DecodeRow` call `DecodeRoWIthMap`.

Reply:
Got it.